### ISSUE TITLE
iOS (Cocopaods) dependency update - GoogleDataTransport

### ISSUE THREAD

**Issue by Truba** [flutter-ml/google_ml_kit_flutter#668]
Could you 🙏  update dependency in `MLKitCommon` for `GoogleDataTransport` to version 10+.
Firebase seems to be using newer version.

The error:
```
[!] CocoaPods could not find compatible versions for pod "GoogleDataTransport":
  In Podfile:
    firebase_messaging (from `.symlinks/plugins/firebase_messaging/ios`) was resolved to 15.1.0, which depends on
      Firebase/Messaging (= 11.0.0) was resolved to 11.0.0, which depends on
        FirebaseMessaging (~> 11.0.0) was resolved to 11.0.0, which depends on
          GoogleDataTransport (~> 10.0)

     GoogleMLKit/BarcodeScanning (~> 6.0.0) was resolved to 6.0.0, which depends on
        MLKitBarcodeScanning (~> 5.0.0) was resolved to 5.0.0, which depends on
          MLKitCommon (~> 11.0) was resolved to 11.0.0, which depends on
            GoogleDataTransport (< 10.0, >= 9.4.1)
```

**Comment by AndreyDAraya** [comment]
The fix, I talk with Google member and recommend this actions:

https://github.com/google/GoogleDataTransport/issues/149#issuecomment-2313898323

firebase_core: ^2.32.0
firebase_remote_config: ^4.4.7
firebase_analytics: ^10.10.7
...Other packages

Or the version old 3 month.

**Comment by bensonarafat** [comment]
duplicate #669

**Comment by dipakravalVH** [comment]
[!] CocoaPods could not find compatible versions for pod "GoogleDataTransport":
  In Podfile:
    firebase_messaging (from `.symlinks/plugins/firebase_messaging/ios`) was resolved to 15.1.0, which depends on
      Firebase/Messaging (= 11.0.0) was resolved to 11.0.0, which depends on
        FirebaseMessaging (~> 11.0.0) was resolved to 11.0.0, which depends on
          GoogleDataTransport (~> 10.0)

    google_mlkit_face_detection (from `.symlinks/plugins/google_mlkit_face_detection/ios`) was resolved to 0.11.0, which depends on
      GoogleMLKit/FaceDetection (~> 6.0.0) was resolved to 6.0.0, which depends on
        MLKitFaceDetection (~> 5.0.0) was resolved to 5.0.0, which depends on
          MLKitCommon (~> 11.0) was resolved to 11.0.0, which depends on
            GoogleDataTransport (< 10.0, >= 9.4.1)
            
            
Same error with latest version:

firebase_messaging: ^15.1.0
firebase_analytics: ^11.3.0
firebase_crashlytics: ^4.1.0
firebase_core: ^3.4.0
